src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -104,14 +104,8 @@ private LocalDefinition EmitAddress(BoundExpression expression, AddressKind addr
                     break;
 
                 case BoundKind.AssignmentOperator:
-                    var assignment = (BoundAssignmentOperator)expression;
-                    if (assignment.RefKind == RefKind.None)
-                    {
-                        goto default;
-                    }
-
-                    EmitAssignmentExpression(assignment, UseKind.UsedAsAddress);
-                    break;
+                    Debug.Assert(((BoundAssignmentOperator)expression).RefKind == RefKind.None);
```

Comment:
We never lower declarations to assignments? Also, do we want to be stronger than an Assert here? If we have a bug this could generate bad code, right?